src/SprykerMiddleware/Zed/Process/Communication/Plugin/Stream/XmlInputStreamPlugin.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerMiddleware\Zed\Process\Communication\Plugin\Stream;
+
+use SprykerMiddleware\Shared\Process\Stream\ReadStreamInterface;
+use SprykerMiddleware\Zed\Process\Communication\Plugin\StreamConfigurator\StreamConfiguratorInterface;
+use SprykerMiddleware\Zed\Process\Communication\Plugin\StreamConfigurator\XmlInputStreamConfigurator;
+use SprykerMiddleware\Zed\Process\Dependency\Plugin\Stream\InputStreamPluginInterface;
+
+/**
+ * @method \SprykerMiddleware\Zed\Process\Business\ProcessFacadeInterface getFacade()
+ * @method \SprykerMiddleware\Zed\Process\Communication\ProcessCommunicationFactory getFactory()
+ */
+class XmlInputStreamPlugin extends AbstractOptionAwareStreamPlugin implements InputStreamPluginInterface
+{
+    protected const PLUGIN_NAME = 'XmlInputStreamPlugin';
+
+    /**
+     * @api
+     *
+     * @return string
+     */
+    public function getName(): string
+    {
+        return static::PLUGIN_NAME;
+    }
+
+    /**
+     * @api
+     *
+     * @param string $path
+     *
+     * @return \SprykerMiddleware\Shared\Process\Stream\ReadStreamInterface
+     */
+    public function getInputStream(string $path): ReadStreamInterface
+    {
+        return $this->getFactory()
+            ->createStreamFactory()
```

Comment:
This is a common approach here.

It looks like a violation of our conventions in Middleware in general, because we need to create an object from the business layer in the communication layer, and for this was introduced here special stream factory which anyway created directly in communication factory in spite of its located in the business layer.

I guess this question could be considered out of this bug.

Reply:
ok